### PR TITLE
New module elasticjob-restful and refactor cloud-scheduler with elasticjob-restful

### LINKED ISSUE #1284: Looking for a comfortable command format for cloud RESTful API refactoring
## Discussion
Links #1193 .

According to my statistics, there are 25 APIs in elasticjob-cloud-scheduler need to be refactored.
We are going to replace RESTful API with socket to keep the command style consistent with SnapshotService.
Now we need to redefine the following APIs in a format like `dump@jobName`, so that we can send request in the way like:
```bash
echo "dump@jobName" | nc localhost 9888
```

CloudAppController:
- POST /app
- PUT /app
- GET /app/list
- GET /app/{appName}
- GET /app/{appName}/disable
- POST /app/{appName}/disable
- POST /app/{appName}/enable
- DELETE /app/{appName}

CloudJobController:
- POST /job/register
- PUT /job/update
- DELETE /job/{jobName}/deregister
- GET /job/{jobName}/disable
- POST /job/{jobName}/disable
- POST /job/{jobName}/enable
- POST /job/trigger
- GET /job/jobs
- GET /job/jobs/{jobName}
- GET /job/tasks/running
- GET /job/tasks/ready
- GET /job/tasks/failover
- GET /job/events/executions
- GET /job/events/statusTraces

CloudOperationController:
- POST /operate/reconcile/explicit
- POST /operate/reconcile/implicit
- GET /operate/sandbox

There is something needs to be discussed:
1. How to express HTTP methods and URL in a comfortable command line format?
2. What format do we use to attach a request body to the command?
3. What format do we use to attach authentication information to the command?

If you have any suggestions at all, whether large or small, please leave your comments.

### PR DESCRIPTION
**Pull request by TeslaCN**
Fixes #1284 

Changes proposed in this pull request:
- New module elasticjob-restful.
- Refactor cloud-scheduler with elasticjob-restful.

### CODE REVIEW

File: elasticjob-cloud/elasticjob-cloud-scheduler/src/main/java/org/apache/shardingsphere/elasticjob/cloud/console/config/serializer/JsonResponseBodySerializer.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.google.gson.GsonBuilder;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import org.apache.shardingsphere.elasticjob.restful.serializer.ResponseBodySerializer;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Json response body serializer. Serialize object to json byte except String.
+ */
+public class JsonResponseBodySerializer implements ResponseBodySerializer {
```
Comment (Technoboy-) [inline elasticjob-cloud/elasticjob-cloud-scheduler/src/main/java/org/apache/shardingsphere/elasticjob/cloud/console/config/serializer/JsonResponseBodySerializer.java]: Miss final

Comment (TeslaCN) [reply elasticjob-cloud/elasticjob-cloud-scheduler/src/main/java/org/apache/shardingsphere/elasticjob/cloud/console/config/serializer/JsonResponseBodySerializer.java]: Resolved in #1417

File: elasticjob-cloud/elasticjob-cloud-scheduler/src/main/java/org/apache/shardingsphere/elasticjob/cloud/console/controller/CloudAppController.java [hunk 2]
```diff
@@ -33,21 +31,22 @@
...
+import java.util.Collection;
+import java.util.Optional;
 
 /**
  * Cloud app controller.
  */
-@RestController
-@RequestMapping("/app")
-public final class CloudAppController {
+@ContextPath("/api/app")
```
Comment (Technoboy-) [inline elasticjob-cloud/elasticjob-cloud-scheduler/src/main/java/org/apache/shardingsphere/elasticjob/cloud/console/controller/CloudAppController.java:48]: If we add /api, do we need to modify the front?

Comment (TeslaCN) [reply elasticjob-cloud/elasticjob-cloud-scheduler/src/main/java/org/apache/shardingsphere/elasticjob/cloud/console/controller/CloudAppController.java:48]: UI has been separated. The prefix /api was existed before and I didn't change the path.

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**